lib/coupon_code.rb

```diff
@@ -19,19 +19,27 @@ def self.generate(options = { parts: PARTS })
     parts.join('-')
   end
 
-  def self.validate(orig, num_parts = PARTS)
-    code = orig.upcase
-    code.gsub!(/[^0-9A-Z]+/, '')
-    parts = code.scan(/[0-9A-Z]{#{LENGTH}}/)
+  def self.validate(orig, options = { parts: @@parts })
+    num_parts = options.delete(:parts)
+    code      = orig.upcase.gsub(/[^0-9A-Z]+/, '')
```

Comment:
Unnecessary spacing detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.